Godeps/_workspace/src/github.com/docker/distribution/registry/storage/layerwriter.go

```diff
@@ -298,8 +336,13 @@ func (lw *layerWriter) validateLayer(dgst digest.Digest) (digest.Digest, error)
 	)
 
 	if lw.resumableDigester != nil {
+		offset := lw.size
+		if offset < lw.offset {
+			// we're probably on NFS where file attributes are cached on client
+			offset = lw.offset
```

Comment:
this is also concerning... if our sizes don't match, we silently update to an expected offset? this seems likely to cause corruption